apps/shell/tash_security.c

```diff
@@ -23,43 +23,129 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <mbedtls/sha256.h>
+#include <mbedtls/pkcs5.h>
+#include <security/security_crypto.h>
 #include "tash_internal.h"
 
-#define SHA256_HASH_nCHAR  (64) // 32 bytes
-#define SHA256_HASH_nBYTE  (32) // 32 bytes
-#define CONVERT_nCHAR      (sizeof(unsigned long) * 2) // unsigned long for strtoul
+#define CONVERT_nCHAR			(sizeof(unsigned long) * 2) // unsigned long for strtoul
+
+#define PBKDF2_SALT_ENCRYPTED	"\x50\x3B\x3C\x3E\xBF\x0D\x3D\x2C\x9D\x6E\xA2\xF3\x62\xF5\x0F\x9F"
```

Comment:
This and the iteration value could be changed by user. Let's provide them as configs.